src/Krizalys/Onedrive/Client.php

```diff
@@ -618,7 +619,7 @@ public function createFile($name, $parentId = null, $content = '')
 
         // TODO: some versions of cURL cannot PUT memory streams? See here for a
         // workaround: https://bugs.php.net/bug.php?id=43468
-        $file = $this->apiPut($parentId . '/files/' . urlencode($name), $stream);
+        $file = $this->apiPut($parentId . '/files/' . urlencode($name) . '?overwrite=' . ($overwrite ? 'true' : 'ChooseNewName'), $stream);
```

Comment:
Apparently, passing `overwrite=ChooseNewName` does not prevent the file to be overwritten. Passing `overwrite=false` does, however.

This feature does not seem to be documented, where did you see `overwrite=ChooseNewName` actually working as you suggested?

Using `overwrite=false` sounds like a safer (working) solution to me, plus it's probably the value that most users would expect.